load-tests/src/main.rs

```diff
@@ -1,25 +1,160 @@
-use std::time::Duration;
+mod constants;
+mod primitives;
+pub mod utils;
 
-use goose::prelude::*;
-use goose_eggs::{validate_and_load_static_assets, Validate};
+use core::panic;
+use log::info;
 use reqwest::{header::CONTENT_TYPE, Body};
+use std::{time::Duration, vec};
+
+use constants::VALID_OIDC_PROVIDER_KEY;
+use goose::prelude::*;
+use mpc_recovery::{
+    msg::{
+        ClaimOidcRequest, MpcPkRequest, NewAccountRequest, SignRequest, UserCredentialsRequest,
+        UserCredentialsResponse,
+    },
+    sign_node::oidc::OidcToken,
+    transaction::CreateAccountOptions,
+    utils::{
+        claim_oidc_request_digest, sign_digest, sign_request_digest,
+        user_credentials_request_digest,
+    },
+};
+use near_crypto::SecretKey;
+use near_primitives::{
+    account::{AccessKey, AccessKeyPermission},
+    borsh::BorshSerialize,
+    delegate_action::DelegateAction,
+    transaction::{Action, AddKeyAction},
+    types::AccountId,
+};
+use primitives::UserSession;
+use rand::{distributions::Alphanumeric, Rng};
+use utils::build_send_and_check_request;
 
 #[tokio::main]
 async fn main() -> Result<(), GooseError> {
+    env_logger::init();
     GooseAttack::initialize()?
         .register_scenario(
-            scenario!("simple_mpc_public_key").register_transaction(transaction!(mpc_public_key)),
+            scenario!("registration")
+                .register_transaction(transaction!(prepare_user_credentials).set_sequence(1))
+                .register_transaction(transaction!(claim_oidc).set_sequence(2))
+                .register_transaction(transaction!(new_account).set_sequence(3)),
+        )
+        .register_scenario(
+            scenario!("registrationAndSign")
+                .register_transaction(transaction!(prepare_user_credentials).set_sequence(1))
+                .register_transaction(transaction!(claim_oidc).set_sequence(2))
+                .register_transaction(transaction!(new_account).set_sequence(3))
+                .register_transaction(transaction!(user_credentials).set_sequence(4))
+                .register_transaction(
+                    transaction!(sign)
+                        .set_sequence(5)
+                        .set_weight(1000) // In this scenario we are mostly testing /sign functionality
```

Comment:
What does `set_weight` actually do here? I would assume that these 5 actions above need to be executed in sequence, so why would we prioritize the last step over the others?

Reply:
In the terminal, you are passing a number of users (N) and a scenario (`registrationAndSign` in this case). Then each user executes transactions in sequence (`check .set_sequence(k)`). The scenario starts from the beginning after the user ends the previous one.

In these particular load tests we are trying to make as many sign requests as possible. Without `.set_weight(1000)` sign will be executed once per registration and registration takes several seconds.